torch/_inductor/codegen/wrapper.py

```diff
@@ -85,10 +86,16 @@ class EnterCudaDeviceContextManagerLine:
     def codegen(self, code: IndentedBuffer, device_cm_stack: contextlib.ExitStack):
         if V.graph.cpp_wrapper:
             code.writeline("\n")
-            if self.first_time:
-                code.writeline(f"at::cuda::CUDAGuard device_guard({self.device_idx});")
+            if torch.version.hip is None:
+                if self.first_time:
+                    code.writeline(f"at::cuda::CUDAGuard device_guard({self.device_idx});")
+                else:
+                    code.writeline(f"device_guard.set_index({self.device_idx});")
```

Comment:
The device guard looks to be the same for both hip and cuda?  could we just have if self.first_time: then check for hip vs cuda?

Reply:
Ah thanks for spotting that, this is a mistake that I brought in when copying to this branch.

We need to use HIPGuard
`code.writeline(f"at::hip::HIPGuard device_guard({self.device_idx});")`